crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -252,6 +252,9 @@ impl<'w, 's, 'a> ChildBuilder<'w, 's, 'a> {
 
 /// Trait defining how to build children
 pub trait BuildChildren {
+    /// Spawns a new entity for the provided `bundle`, and adds it as a child entity to the parent
+    fn with_child(&mut self, bundle: impl Bundle) -> &mut Self;
```

Comment:
I'd be inclined to make this return `()`, as this could reasonably return either `Self` or a version of `Self` for the new entity (especially as there's no way in this API to get the new child's id directly.

Reply:
I think users would get that it's returning the child_builder cause that's a common pattern used in Bevy.

Reply:
Returning self is actually quite useful: you can spawn multiple children under the same parent by chaining it. I should improve the docs to explain that though.

I didn't change the examples to use that pattern though, as it would be pointlessly controversial.

Reply:
Yeah, I was a bit confused by that too. I think it makes sense as it is and overall is more useful this way, but I do think there's some potential for confusion here.

Reply:
Yep, I also wanted to maintain consistency with with_children

Reply:
If #6886 can be made to work and is adopted,  it really needs this to be usable and chaining multiple children probably as well.